detectors/aws/ecs/ecs.go

```diff
@@ -43,27 +43,32 @@ var (
 )
 
 // Create interface for methods needing to be mocked
-type detectorUtils interface {
+type detectorUtilsResources interface {
 	getContainerName() (string, error)
 	getContainerID() (string, error)
 }
 
-// struct implements detectorUtils interface
-type ecsDetectorUtils struct{}
+// struct implements detectorUtilsResources interface
+type DetectorUtils struct{}
 
 // resource detector collects resource information from Elastic Container Service environment
-type ResourceDetector struct {
-	utils detectorUtils
+type resourceDetector struct {
+	utils detectorUtilsResources
 }
 
-// compile time assertion that ecsDetectorUtils implements detectorUtils interface
-var _ detectorUtils = (*ecsDetectorUtils)(nil)
+// compile time assertion that ecsDetectorUtils implements detectorUtilsResources interface
+var _ detectorUtilsResources = (*DetectorUtils)(nil)
 
 // compile time assertion that resource detector implements the resource.Detector interface.
-var _ resource.Detector = (*ResourceDetector)(nil)
+var _ resource.Detector = (*resourceDetector)(nil)
+
+// returns resource detector struct
```

Comment:
This comment needs to follow Go documentation guidelines (i.e. start with the function name)

```suggestion
// NewResourceDetector returns a resource detector that will detect AWS ECS resources.
```